zhaquirks/xiaomi/aqara/motion_agl1.py

```diff
@@ -0,0 +1,194 @@
+
+
+class OppleCluster(XiaomiAqaraE1Cluster):
+    """Aqara manufacturer cluster for the presence sensor FP1E."""
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Manufacturer specific attributes."""
+
+        # The configurable maximum detection distance in millimeters (default 600 = 6 meters).
+        approach_distance = ZCLAttributeDef(
+            id=0x015B,
+            type=types.uint32_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        # Detected motion
+        motion = ZCLAttributeDef(
+            id=0x0160,
+            type=AqaraMotion,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        # Distance to the detected motion in millimeters
+        motion_distance = ZCLAttributeDef(
+            id=0x015F,
+            type=types.uint32_t,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        # The configurable detection sensitivity
+        motion_sensitivity = ZCLAttributeDef(
+            id=0x010C,
+            type=AqaraMotionSensitivity,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        # Detected occupancy
+        occupancy = ZCLAttributeDef(
+            id=0x0142,
+            type=AqaraOccupancy,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        # Trigger AI spatial learning (write 1)
+        reset_no_presence_status = ZCLAttributeDef(
+            id=0x0157,
+            type=types.uint8_t,
+            access="w",
+            is_manufacturer_specific=True,
+        )
+
+        # Trigger device restart (write 0)
+        restart_device = ZCLAttributeDef(
+            id=0x00E8,
+            type=types.Bool,
+            access="w",
+            is_manufacturer_specific=True,
+        )
+
+    def _update_attribute(self, attrid: int, value: Any) -> None:
+        super()._update_attribute(attrid, value)
+        if attrid == self.AttributeDefs.occupancy.id:
+            self.endpoint.occupancy.update_attribute(
+                OccupancySensing.AttributeDefs.occupancy.id,
+                OccupancySensing.Occupancy.Occupied
+                if value == AqaraOccupancy.Occupied
+                else OccupancySensing.Occupancy.Unoccupied,
+            )
+        elif attrid == self.AttributeDefs.motion.id:
+            self.endpoint.ias_zone.update_attribute(
+                IasZone.AttributeDefs.zone_status.id,
+                IasZone.ZoneStatus.Alarm_1 if value == AqaraMotion.Moving else 0,
+            )
+
+
+(
+    QuirkBuilder("aqara", "lumi.sensor_occupy.agl1")
+    .friendly_name(model="Presence Sensor FP1E", manufacturer="Aqara")
+    .adds(DeviceTemperature)
+    .adds(OccupancySensingLocal)
+    .adds(
+        IasZoneLocal,
+        constant_attributes={
+            IasZone.AttributeDefs.zone_type: IasZone.ZoneType.Motion_Sensor
+        },
```

Comment:
I think I remember someone complaining about `constant_attributes` defined via quirks v2 not always working as expected. And I might see why: zigpy adds the attribute name as the key, not the attribute id. This might work in some places, but is unexpected in others. Zigpy code for reference: [zigpy/quirks/v2/__init__.py#L178-L182](https://github.com/zigpy/zigpy/blob/8e9f5163ef93ff71fa0ce4992076d8bc248eac44/zigpy/quirks/v2/__init__.py#L178-L182)

Until this is fixed in zigpy, let's remove this here and add it in the cluster class above via the attribute id in `_CONSTANT_ATTRIBUTES`. We're defining `_VALID_ATTRIBUTES` there anyway, so they're also together then.
We'll want to update this again when we add valid attributes as a parameters here for quirks v2 (and fix the constant attributes issue).
```suggestion
```

Reply:
Good catch, thanks for making that change.